mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -598,6 +606,10 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
           targetGemmDim2Attr.push_back(b.getNamedAttr(
               "names", b.getArrayAttr({b.getStringAttr(gemmKPad_name)})));
         }
+        // filter of forward, gemmK=c*y*x
+        filterOobCheckDims.insert(nameToDims["c"]);
+        filterOobCheckDims.insert(nameToDims["y"]);
```

Comment:
@ltqin can we use `needExtraPad` and `gemmKExtra` to help us understand should `c/y/x` all be inserted, or just `c`?

Reply:
for gkcxy data layout, if  y =( UpIndex % (Y*X))/X and x = ( UpIndex % (Y*X))%X , I think it not necessary to oob check y and x.
yes ,"needExtraPad" and "gemmKExtra" can help us understand that  in different case in source code